apps/cowswap-frontend/src/modules/trade/utils/swapErrorHelper.ts

```diff
@@ -1,11 +1,25 @@
 import { getProviderErrorMessage, isRejectRequestProviderError } from '@cowprotocol/common-utils'
 
+import { getIsOrderBookTypedError } from 'api/gnosisProtocol'
+
 export const USER_SWAP_REJECTED_ERROR = 'User rejected signing the order'
 
 export function getSwapErrorMessage(error: Error): string {
   if (isRejectRequestProviderError(error)) {
     return USER_SWAP_REJECTED_ERROR
   } else {
-    return getProviderErrorMessage(error)
+    const defaultErrorMessage = getProviderErrorMessage(error)
+
+    if (getIsOrderBookTypedError(error)) {
+      return capitalizeFirstLetter(error.body?.description) || defaultErrorMessage
+    }
+
+    return defaultErrorMessage
   }
 }
+
+function capitalizeFirstLetter(str: string): string {
```

Comment:
nit: i would move this to a more general lib

Reply:
Done